### ISSUE TITLE
Add `ArrayNode#set(int index, primitive_type value)`

### ISSUE THREAD

**Issue by TarekkMA** [FasterXML/jackson-databind#3154]
**Is your feature request related to a problem? Please describe.**
Currently, we could `add`/`insert` values to `ArrayNode` instance. but the `set` method only accepts `JsonNode` instance. 

**Describe the solution you'd like**
It would be nice if we have `set` methods like `add`/`insert` 

**Usage example**
```java
ArrayNode arr = getNewArrayNode();
arr.add("a");
arr.insert(0, "b");
arr.set(0, "c");        // <------- Missing
```
**Additional context**
N/A

**Comment by TarekkMA** [comment]
If this is accepted, I will be happy to make a PR for it.

**Comment by cowtowncoder** [comment]
Sounds reasonable. I think it should probably follow example of `ObjectNode.put()` so that:

1. A wide variety of types are supported (and esp. important is primitive-and-wrapper variants)
2. Returns `ObjectNode` (`this`) to allow call chaining.

It might also make sense to post on `jackson-dev` google group to ask for input: this makes sense to me but others might be able to offer comments.

Also: needs to go in `2.13` branch, being feature add.

**Comment by TarekkMA** [comment]
Thank you for the feedback I will post this issue on the google group for more comments about it.

**Comment by cowtowncoder** [comment]
Thank you again @TarekkMA for contributing this! It will be included in 2.13.0 once that is ready (hoping to get release candidate out during June 2021).
